modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/builder/ConfigurationSchemaDelegate.java

```diff
@@ -38,25 +43,48 @@
   private final SchemaBuilder builder;
   private Schema schema;
 
-  public ConfigurationSchemaDelegate(SchemaBuilder builder) {
+  ConfigurationSchemaDelegate(SchemaBuilder builder) {
     this.builder = builder;
   }
 
-  public void registerConfigElement(Schema schema, final ConfigurationModel configurationModel,
-                                    DslElementSyntax dslConfigElement) {
+  void registerConfigElement(Schema schema, final ConfigurationModel configurationModel,
+                             DslElementSyntax dslConfigElement) {
     this.schema = schema;
 
     ExtensionType config = registerExtension(dslConfigElement.getElementName());
     config.getAttributeOrAttributeGroup().add(builder.createNameAttribute(true));
+    config.setAnnotation(builder.createDocAnnotation(configurationModel.getDescription()));
 
-    final ExplicitGroup choice = new ExplicitGroup();
-    choice.setMinOccurs(ZERO);
-    choice.setMaxOccurs(UNBOUNDED);
+    Optional<TopLevelElement> connectionElement = addConnectionProviderElement(configurationModel);
+    Optional<TopLevelElement> policyElement = addDynamicConfigPolicyElement(configurationModel);
 
-    addConnectionProviderElement(choice, configurationModel);
-    addDynamicConfigPolicyElement(choice, configurationModel);
-    builder.registerParameters(config, choice, configurationModel.getAllParameterModels());
-    config.setAnnotation(builder.createDocAnnotation(configurationModel.getDescription()));
+    Map<String, Map<String, TopLevelElement>> groups = new LinkedHashMap<>();
+    configurationModel.getParameterGroupModels()
```

Comment:
@marianogonzalez please check this makes sense for you, since is an example of the rules for sorting the parameters in the schema